analysis/quantifier-comparisons/01-import-quant-data.Rmd

```diff
@@ -0,0 +1,235 @@
+---
+title: "SCRNAseq quantification comparisons: Data Import"
+author: "Joshua Shapiro for CCDL"
+output: 
+  html_notebook:
+    toc: true
+    toc_float: true
+---
+
+This notebook contains code to import and do preliminary processing single cell expression data, quantified by Alevin, Kallisto, and Cellranger. 
+Alevin samples were quantified with and without decoy sequences. 
+The imported data is stored as SingleCellExperiment objects, which are then saved as an `.rds` file for later analysis.
+
+## Setup
+
+### Load Libraries for import
+```{r setup}
+library(magrittr)
+library(SingleCellExperiment)
+library(DropletUtils)
+library(tximport)
+```
+
+
+### File and directory setup
+
+Input file locations first, local and remote (S3)
+```{r}
+base_dir <- here::here()
+
+data_dir <- file.path(base_dir, 'data', 'quants')
+dir.create(data_dir, recursive = TRUE, showWarnings = FALSE)
+
+alevin_data_dir <- file.path(data_dir, 'alevin')
+dir.create(alevin_data_dir, recursive = TRUE, showWarnings = FALSE)
+quant_s3_alevin <- 's3://nextflow-ccdl-results/scpca/alevin-quant'
+
+kallisto_data_dir <- file.path(data_dir, 'kallisto')
+dir.create(kallisto_data_dir, recursive = TRUE, showWarnings = FALSE)
+quant_s3_kallisto <- 's3://nextflow-ccdl-results/scpca/kallisto-quant'
+
+cellranger_data_dir <- file.path(data_dir, 'cellranger')
+dir.create(cellranger_data_dir, recursive = TRUE, showWarnings = FALSE)
+quant_s3_cellranger <- 's3://nextflow-ccdl-results/scpca/cellranger-quant'
+```
+
+Output files will be a TSV format info file and `.rds` files of the SingleCellExperiment objects, stored in a subdirectory of this notebook's location.
+
+```{r}
+export_dir <- 'data'
+if (!dir.exists(export_dir)){
+  dir.create(export_dir)
+}
+
+quant_info_file <- file.path(export_dir, 'quant_info.tsv')
+
+alevin_rds <- file.path(export_dir, 'alevin_sces.rds')
+kallisto_rds <- file.path(export_dir, 'kallisto_sces.rds')
+cellranger_rds <- file.path(export_dir, 'cellranger_sces.rds')
+
+```
+
+
+
+### Sync S3 files
+
+```{r}
+samples = c('SCPCR000001', 'SCPCR000002')
+
+# generate include statements for sample directories
+includes = stringr::str_glue("--include \"*/{samples}*\"")
```

Comment:
Why `=` assignment?

Reply:
Got me! My best guess is that this was because I was going back and forth between R and nextflow when I wrote this.